src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -764,14 +769,57 @@ private void cut() {
      *            "deleted". If true the action will be localized as "cut"
      */
     private void delete(boolean cut) {
```

Comment:
In the end, you could probably extract parts of this method into separate methods or maybe even extract a class here and move that into `org.jabref.gui.groups`. I am a bit wary about `BasePanel` growing even more. The class is already far too big.